Framework/src/QcInfoLogger.cxx

```diff
@@ -97,11 +96,12 @@ void QcInfoLogger::init(const std::string& facility,
                         int run,
                         const std::string& partitionName)
 {
+  DiscardFileParameters discardFileParameters;
   std::string discardDebugStr = config.get<std::string>("qc.config.infologger.filterDiscardDebug", "false");
-  bool discardDebug = discardDebugStr == "true";
-  int discardLevel = config.get<int>("qc.config.infologger.filterDiscardLevel", 21 /* Discard Trace */);
-  std::string discardToFile = config.get<std::string>("qc.config.infologger.filterDiscardFile", "");
-  init(facility, discardDebug, discardLevel, discardToFile, dplInfoLogger, dplContext, run, partitionName);
+  discardFileParameters.debug = discardDebugStr == "true";
+  discardFileParameters.fromLevel = config.get<int>("qc.config.infologger.filterDiscardLevel", 21 /* Discard Trace */);
+  discardFileParameters.discardFile = config.get<std::string>("qc.config.infologger.filterDiscardFile", "");
+  init(facility, discardFileParameters, dplInfoLogger, dplContext, run, partitionName);
```

Comment:
why are you not reloading all the parameters?

Reply:
good point, will fix.